reddit.js

```diff
@@ -0,0 +1,205 @@
+{
+	"translatorID": "23bacc11-98e3-4b78-b1ef-cc2c9a04b893",
+	"label": "reddit",
+	"creator": "Lukas Kawerau",
+	"target": "^https?://www\\.reddit\\.com",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-10-29 16:05:25"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+	Copyright © 2020 Lukas Kawerau
+	This file is part of Zotero.
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+	***** END LICENSE BLOCK *****
+*/
+
+
+// attr()/text() v2
+// eslint-disable-next-line
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null}
+
+function detectWeb(doc, url) {
+	var regex = new RegExp("\\/r\\/[a-z\\d]+\\/comments\\/");
+	if (url.includes('search') && getSearchResults(doc, true)) {
+		return 'multiple';
+	} 
+	else if (regex.test(url)) {
+		return 'forumPost';
+	}  else {
+		return 'multiple';
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//a[div/h3]');
+	for (let row of rows) {
+		var href = row.href + '.json';
+		var title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	var jsonUrl = url.split("?")[0] + '.json';
+	var commentRegex = new RegExp("\\/r\\/[a-z\\d]+\\/comments\\/[a-z\\d]+\\/[a-z\\d_]+\\/[a-z\\d]+\\/");
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
```

Comment:
```suggestion
				return;
```

for consistent return